packages/iov-keycontrol/src/proposal.ts

```diff
@@ -0,0 +1,176 @@
+    path: ReadonlyArray<Slip0010RawIndex>,
+  ) => Promise<ReadonlyArray<Slip0010RawIndex>>;
+
+  // here you can add anything you want, for just bip44 compliant coins.
+  // to make it more useful, we add many helpers below.
+  //
+  // generate a new HD keyring identity with a given path
+  // a keyring entry may support many possible paths, or only one.
+  // if path or algorithm is not supported by the implementation,
+  // it will throw an error
+  readonly createBip32Identity: (
+    algo: Algorithm,
+    path: ReadonlyArray<Slip0010RawIndex>,
+  ) => Promise<LocalHDIdentity>;
+}
+
+const purpose44 = Slip0010RawIndex.hardened(44);
+
+// purpose will be hardcoded 44' in this case
+// exactly 5 sections in the path
+export const bip44 = (keyring: SlipKeyringEntry) => (
+  algo: Algorithm,
+  coinType: Slip0010RawIndex,
+  account: Slip0010RawIndex,
+  change: Slip0010RawIndex,
+  addressIndex: Slip0010RawIndex,
+): Promise<LocalHDIdentity> =>
+  keyring.createBip32Identity(algo, [purpose44, coinType, account, change, addressIndex]);
+
+// bip44Auto will auto-generate an appropriate addressIndex
+export const bip44Auto = (keyring: SlipKeyringEntry) => async (
+  algo: Algorithm,
+  coinType: Slip0010RawIndex,
+  account: Slip0010RawIndex,
+  change: Slip0010RawIndex,
+): Promise<LocalHDIdentity> => {
+  const zero = algo === Algorithm.SECP256K1 ? Slip0010RawIndex.normal(0) : Slip0010RawIndex.hardened(0);
+  const path = [purpose44, coinType, account, change, zero];
+  const available = await keyring.incrementBip32Address(algo, path);
+  const ident = await keyring.createBip32Identity(algo, available);
+  return ident;
+};
+
+// Keypair is a private and public key for the given algorithm
+export interface Keypair {
+  readonly algo: Algorithm;
+  readonly public: PublicKeyBytes;
+  readonly private: PrivateKeyBytes;
+}
+
+export interface MixedKeyringEntry extends KeyringEntry<LocalIdentity> {
+  readonly kind: KeyringType.Mixed;
+
+  // randomly generate a new private-public key pair
+  readonly createRandomIdentity: (algo: Algorithm) => Promise<LocalIdentity>;
+
+  // import existing public/private key pair
+  readonly importIdentity: (keypair: Keypair) => Promise<LocalIdentity>;
+}
+
+// skipping defining load/store/lock or listeners, as they would
+// remain as is.
+export interface UserProfile {
+  // actually, can we make this `addKeyring(entry: Keyring)`
+  // I don't think the Entry word adds clarity....
+  readonly addEntry: (entry: KeyringEntry) => void;
+
+  // this changes upon addEntry, watch the individual entries
+  // for changes on their state.
+  //
+  // Note that this assumes the caller is trusted as we expose the
+  // full entries (including ability to create identities).
+  //
+  // Also, it will not work well over proxies as it stands (which
+  // should be fixed shortly).
+  readonly getEntries: () => ValueAndUpdates<ReadonlyArray<KeyringEntry>>;
+
+  // setEntryLabel should be called on the Entry itself
```

Comment:
i think this name is clearer than the one in use (`setLabel`) but noticed that specific name also doesn't appear in the code so am confused by it.

Reply:
proposal.ts is just a syntax-correct suggestion of the api, no real code.
once the proposal is accepted, then another pr would be to update the code to match